### PR TITLE
Jobs table fix

### PR DESCRIPTION
**Pull request by swainn**
The jobs table was not updating as it should likely as a result of changes made during the development of the Dask job. The behavior of jobs that were Pending seems to have changed or been fixed. The primary fix in the PR is changing statuses from Pending to Submitted when a job is executed programatically.

### CODE REVIEW

**Comment by swainn** [comment]
This addresses #384

**Comment by swainn** [comment]
Addresses #405

**Comment by swainn** [comment]
@SarvaPulla and @sdc50 this should be good to go. Included in this pull request:
* Addressed several issues that were causing the jobs table not to update including
    * PEN -> SUB when using execute() method
    * Use django timezones for datetimes in jobs models
* Addressed instabilities with the Dask Jobs API
    * Submitting many Dasks jobs would cause Tethys to lock up or crash
    * Dask distributed clients were not being closed properly
* Cleans up the Jobs table for Dask jobs showing only one instance of the Dask Task status at the top of the table instead of repeating it for each job.
* Mocked out new dependencies in the docs conf.py (we don't install the dependencies for the docs builds)
* Fixed a few hold-over docs typos for spatial dataset services.

File: tethys_gizmos/static/tethys_gizmos/js/jobs_table.js [hunk 1]
```diff
@@ -204,14 +203,12 @@ function update_row(table_elem){
             });
             status = json.status;
             if(status == 'Running' || status == 'Submitted' || status == 'Various'){
-                var run_time_field = $('#run_time-' + job_id)
+                active_counter++;
```
Comment (sdc50) [inline tethys_gizmos/static/tethys_gizmos/js/jobs_table.js]: Is this being used?

Comment (sdc50) [reply tethys_gizmos/static/tethys_gizmos/js/jobs_table.js]: Got it. Looks like it's a global that gets initialized below.

**Comment by sdc50** [comment]
@SarvaPulla the unofficial "official" policy of Tethys platform is that a PR can be merged as long as it gets Two approvals from the steering committee and no disapprovals. We haven't been great about following those guidelines, but maybe we should start. Since @swainn submitted the PR I think that counts as an implicit approval. I've just reviewed it and approve. Unless you have any objections then I think we should be good to merge this in.

**Comment by SarvaPulla** [comment]
I don't have any objections. Feel free to merge it.

**Pull request merged**